src/Compilers/CSharp/Test/Symbol/Symbols/SymbolEqualityTests.cs

```diff
@@ -765,9 +959,12 @@ private void VerifyEquality(ISymbol type1, ISymbol type2, bool expectedDefault,
             Assert.Equal(expectedIncludeNullability, type2.Equals(type1, SymbolEqualityComparer.IncludeNullability));
 
             // GetHashCode
-            Assert.Equal(type1.GetHashCode(), type2.GetHashCode());
-            Assert.Equal(SymbolEqualityComparer.Default.GetHashCode(type1), SymbolEqualityComparer.Default.GetHashCode(type2));
-            Assert.Equal(SymbolEqualityComparer.IncludeNullability.GetHashCode(type1), SymbolEqualityComparer.IncludeNullability.GetHashCode(type2));
+            if (expectHashCodeMatch)
```

Comment:
>expectHashCodeMatch [](start = 16, length = 19)

Why is this conditional on a separate flag? If ```Object.Equals(Object)``` returns true, the hash code must match, so this should be conditioned on expectedDefault.